### PR TITLE
Make the CI (integration tests) more resilient

### PR DESCRIPTION
**Pull request by eloylp**
**Referenced issue**

Single PR

**Summary of changes**

* Increasing fees for integration testing.
* Added simple exp backoff for contract incokations.

**Reviewer recommendations**

* Check no timeouts in CI.
* Check commits.

### CODE REVIEW

File: contracts/voting/Makefile.toml
Comment (eloylp) [inline contracts/voting/Makefile.toml:33]: This is a custom `cargo-make` plugin implementation of an exponential backoff for retrying transactions. Its based in the `cargo-make` [plugin](https://github.com/sagiegurari/cargo-make#usage-plugins) feature plus, some scripting on [duckscript](https://github.com/sagiegurari/duckscript) :duck: , its native scripting language.

Now we just need to mark the tasks we want to enable the exp retry by adding the `plugin = "retry_script"` attribute to it.


In order to test this, you can clone this branch and adding a dummy task like:

```toml
[tasks.run_retry]
plugin = "retry_script"
script = '''
  echo "running task !"
  exit 1
'''
```

That :point_up:  will output something like:

```bash
Retrying .. waiting 1000 ms before trying again ...
Trying task run_retry - try number 1
Retrying .. waiting 4000 ms before trying again ...
Trying task run_retry - try number 2
Retrying .. waiting 9000 ms before trying again ...
Trying task run_retry - try number 3
Max retries reached !
[cargo-make] ERROR - Error while running plugin: Source: Unknown Line: 29 - Exit with error code: 1
[cargo-make] WARN - Build Failed.
```

Let me know your thoughts @geofmureithi and @mariopil and if positive, i can extend this to rest of the contracts.

Comment (mariopil) [reply contracts/voting/Makefile.toml:33]: Looks nice. So this will retry the task as long as its exit code != 0?

Comment (geofmureithi) [reply contracts/voting/Makefile.toml:33]: Have you tried a delay?

Comment (eloylp) [reply contracts/voting/Makefile.toml:33]: > Looks nice. So this will retry the task as long as its exit code != 0?

Yeah, its not perfect though, it would be nice to detect the `transaction timeout` specific error, so we would be able to differentiate among "retryable" and "non retryable" errors. But probably good for first iteration :stuck_out_tongue:

Comment (eloylp) [reply contracts/voting/Makefile.toml:33]: > Have you tried a delay? This is possibly enough with the exponential backoff.

There is actually a delay , check the `sleep` command in code. Or you mean just a delay and a second retry would be enough ?

**Comment by eloylp** [comment]
Hello @geofmureithi and @mariopil . Added this last commits:

* https://github.com/eigerco/nebula/pull/89/commits/dc7a6ae7ad38c32df3a15197018ed499722a8d39
* https://github.com/eigerco/nebula/pull/89/commits/15fe70f86d6914c8b422364ba92b5548148a2c6c

And also executed twice the CI. Its being sucessful now. I would merge this if you agree and see how it goes. We can make more PRs for adjusting things.

Also, theres is probably another issue. Looks like we are executing twice the tests `on_pull_request` and `on_push` . Probablt we need to separate the current workflow in 2. One for the PRs and the other ones for when the PR is merged in `main`. Both are necessary to ensure changes are well integrated. Maybe i would take inspiration from the alternative proposaed at https://github.com/eigerco/nebula/pull/89 .

image


But i would let that refactor for another PR/issue. Let me know please.

**Comment by eloylp** [comment]
> Yes, let's do it in a separate PR.

Great, should we merge them ? I would need your approval :sweat_smile:

**Review by mariopil — APPROVED** [review]

**Review by geofmureithi — APPROVED** [review]

**Pull request merged**
